pkg/crypto/crypto.go

```diff
@@ -829,14 +829,24 @@ func (ca *CA) MakeServerCertForDuration(hostnames sets.String, lifetime time.Dur
 
 func (ca *CA) EnsureClientCertificate(certFile, keyFile string, u user.Info, expireDays int) (*TLSCertificateConfig, bool, error) {
 	certConfig, err := GetTLSCertificateConfig(certFile, keyFile)
-	if err != nil {
+	if err != nil || subjectChanged(u, certConfig) {
```

Comment:
I believe we should follow the pattern from `EnsureServerCertificate`, which would mean:
1. create a `func GetClientCertificate(certfile, keyfile string, u user.Info) (*TLSCertificateConfig, error)
2. write the client cert equivalent of https://github.com/openshift/library-go/blob/master/pkg/crypto/crypto.go#L759-L768
3. call it here instead of `GetTLSCertificateConfig()`

Reply:
ack, checking to match the pattern